app/src/main/java/com/github/versus/offline/CacheManager.java

```diff
@@ -0,0 +1,65 @@
+package com.github.versus.offline;
+
+import android.content.Context;
+
+import androidx.room.Room;
+
+import com.github.versus.db.DataBaseManager;
+import com.github.versus.posts.Post;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+public final class CacheManager implements DataBaseManager<Post> {
+
+    private final PostDatabase db;
+    private final PostDAO dao;
+    private static CacheManager instance = null;
+
+
+    private CacheManager(Context context){
+        db= Room.databaseBuilder(context.getApplicationContext(),
+                PostDatabase.class, "cache").build();
+        dao= db.activityDao();
+    }
+
+    public static CacheManager getCacheManager(Context context){
+        if(instance==null){
+            instance= new CacheManager(context);
+        }
+        return instance;
+    }
+    @Override
+    public Future<Boolean> insert(Post post) {
```

Comment:
Maybe we can use [CompletableFuture](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/CompletableFuture.html)

Reply:
Modifie that in DataBaseManager, the implementation returns a CompletableFuture anyway